### PR TITLE
Optionally allow annotation name different than LastAppliedConfigAnnotation

### LINKED ISSUE #97: Map field created by kubectl apply removed when manifestival-enabled app manages same object
Because the annotation name `kubectl.kubernetes.io/last-applied-configuration` is hard coded, manifestival changes are always treated as the next canonical version anything previously set by `kubectl apply`. I think of that field as "the set of things that were applied by `kubectl`" and find it confusing that manifestival uses the same field. With the current behavior, I have frustrated cluster admins whose namespace labels (set via `kubectl apply`) are wiped out when my software comes along and applies its own set of labels via manifestival. If they create the labels via some other process, their labels are left unchanged.

Would it be possible to allow devs to change the annotation name via config to something of their choosing so that "the set of things that were applied by `kubectl`" can be different than "the set of things that were applied by myapp"?

### PR DESCRIPTION
**Pull request by joelsmith**
Fixes #97

### CODE REVIEW

**Comment by joelsmith** [comment]
@jcrossley3 could you please take a look when you have a moment?
Thanks!

**Comment by jcrossley3** [comment]
Sorry! I dig it -- looks pretty non-invasive to me, from what I can tell. I appreciate the test and changelog entry -- do you think it's worth adding a blurb in the README, or nah?

File: README.md [hunk 1]
```diff
@@ -262,6 +262,21 @@ option, like so:
...
+`kubectl.kubernetes.io/last-applied-configuration` which is the same name used
+by `kubectl` and applications which use `kubectl` to apply object values. If
+desired, the annotation name can be changed so that Manifestival-applied fields
+won't interfere with non-conflicting, `kubectl`-applied fields (or vice-versa).
+The annotation name is set like this:
+
+```go
+m, _ := NewManifest(path, UseLastAppliedConfigAnnotation("myapp.example.com/last-applied-configuration"))
+```
+
```
Comment (jcrossley3) [inline README.md:279]: Perfect, thanks!

**Review by jcrossley3 — APPROVED** [review]
Great!

**Pull request merged**